docs/guide/docs/infrastructure/docker.md

```diff
@@ -0,0 +1,122 @@
+---
+id: Docker
+title: Deploying with Docker
+---
+
+Docker is a set of platform-as-a-service products that use OS-level virtualization to deliver software in packages called containers.
+
+> For an optimized Docker experience, download [Docker Desktop.](https://www.docker.com/products/docker-desktop)
+
+## Using Remote Duckling & Language Server
+This command will run Botpress within a single container and use the remote Duckling and Language Server hosted by us. You can get the latest `stable` or `nightly` versions on [DockerHub](https://hub.docker.com/r/botpress/server/tags).
+
+> You should **never** use `nightly` versions in production because they are unstable.
+
+```bash
+docker run -d \
+--name botpress \
+-p 3000:3000 \
+-v botpress_data:/botpress/data \
+botpress/server:$TAG
+```
+For instance, you can run an installation as follows:
+
+```bash
+User@DESKTOP-T1ORLFU MINGW64 /c/Botpress/DockerTests
+$ docker run -d \
+> --name myBotpress \
+> -p 3000:3000 \
+> botpress/server:v12_18_2
+Unable to find image 'botpress/server:v12_18_2' locally
+v12_18_2: Pulling from botpress/server
+32802c0cfa4d: Pulling fs layer
+da1315cffa03: Pulling fs layer
+fa83472a3562: Pulling fs layer
+f85999a86bef: Pulling fs layer
+26b3263a34bb: Pulling fs layer
+7e33a4644769: Pulling fs layer
+f85999a86bef: Waiting
+26b3263a34bb: Waiting
+7e33a4644769: Waiting
+da1315cffa03: Download complete
+fa83472a3562: Verifying Checksum
+fa83472a3562: Download complete
+f85999a86bef: Download complete
+32802c0cfa4d: Download complete
+32802c0cfa4d: Pull complete
+da1315cffa03: Pull complete
+fa83472a3562: Pull complete
+f85999a86bef: Pull complete
+26b3263a34bb: Verifying Checksum
+26b3263a34bb: Download complete
+26b3263a34bb: Pull complete
+7e33a4644769: Verifying Checksum
+7e33a4644769: Download complete
+7e33a4644769: Pull complete
+Digest: sha256:798b0fe332c5bb1b707eb62b30c8ed0a4e0609b3c712ee7201c5a7da7be50b7f
+Status: Downloaded newer image for botpress/server:v12_18_2
+bf038c6f84aaeec11773b93a9748bc6732d573a1c115523f1a3d28d20dc06cbe
+```
+You will be able to access your instance of Botpress on the specified mapped port `-p 3000:3000` by visiting  http://localhost:3000/, and your container name will be myBotpress as set at `--name myBotpress`. Your Docker Desktop will reflect the new container as below:
+
+![Botpress Container on Docker](../assets/docker-new-instance.png)
+
+Hosting Duckling and the Language Server is optional. Your Botpress installation will use our hosted services by default.
+
+Choose to either run one of two containers (running two containers is recommended).
+
+### Single Container With All Services
+
+> ⚠️ **Never** run multiple processes inside a single container in production.
```

Comment:
I get this, can you please elaborate anyways? Running duckling and the language server in the same container is not recommended? I thought it to be lightweight enough to be running alongside/inside the language service. Maybe I should revise my production deployment on this...